FinSearchUnified/Bundle/StoreFrontBundle/Gateway/Findologic/CustomFacetGateway.php

```diff
@@ -120,13 +122,33 @@ public function getAllCategoryFacets(ShopContextInterface $context)
      * @param SimpleXMLElement $filters
      *
      * @return array
+     * @throws Zend_Cache_Exception
      */
     private function hydrate(SimpleXMLElement $filters)
     {
         $facets = [];
+        $hasCategoryFacet = false;
+        $hasVendorFacet = false;
 
         foreach ($filters as $filter) {
-            $facets[] = $this->hydrator->hydrateFacet($filter);
+            $facet = $this->hydrator->hydrateFacet($filter);
+            $facetName = $facet->getName();
+
+            if ($facetName === 'vendor') {
+                $hasVendorFacet = true;
+            }
+            if ($facetName === 'cat') {
+                $hasCategoryFacet = true;
+            }
+
+            $facets[] = $facet;
+        }
+
+        if ($hasCategoryFacet === false) {
+            $facets[] = $this->hydrator->hydrateDefaultCategoryFacet();
+        }
+        if ($hasVendorFacet === false) {
```

Comment:
Checking for falseness is enough.

Reply:
Were you able to identify why the default facet method needs to be called twice instead of once? I had mentioned this on youtrack about the issue. From what i think it should pass with an expectation of once but it is passing for exactly(2) instead.